### PR TITLE
Deploy a CronJob to stop alerts from being enabled on osde2e clusters

### PR DESCRIPTION
**Pull request by jewzaam**
https://issues.redhat.com/browse/OSD-4816

### CODE REVIEW

**Comment by jewzaam** [comment]
on hold until internal ci processes are updated to support this, may require some tweaks to the PR

**Comment by yithian** [comment]
seems sane to me.

does the cronjob serviceaccount not need a permission to apply the label?

**Comment by jewzaam** [comment]
@yithian umm yea. did I leave that out? looking.

**Comment by jewzaam** [comment]
I lied, running out of time in the day, will have to pick up next week :(

File: deploy/osde2e-tweaks/production/10-noalerts-osde2e.cronjob.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            - -c
+            - |
+              # Org IDs to apply to are for osde2e: 1aV37K1VQv2zSStwSkdwBNOUBGI, 1MSRrYaelBWudhmVktupE2t7ABy
+              # https://github.com/openshift/ops-sop/blob/master/v4/howto/noalerts.md#hive-production
+              
+              NAMESPACES=()
+
+              # AWS osde2e namespaces:
+              NAMESPACES+=($(oc get accountclaim --all-namespaces -o jsonpath='{.items[?(@.spec.legalEntity.id=="1aV37K1VQv2zSStwSkdwBNOUBGI")].metadata.namespace}'))
+              NAMESPACES+=($(oc get projectclaim --all-namespaces -o jsonpath='{.items[?(@.spec.legalEntity.id=="1aV37K1VQv2zSStwSkdwBNOUBGI")].metadata.namespace}'))
```
Comment (georgettica) [inline deploy/osde2e-tweaks/production/10-noalerts-osde2e.cronjob.yaml:34]: I am sure this is required, but why?

File: deploy/osde2e-tweaks/22-osde2e-noalerts.clusterrole.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  - list
+  - watch
+- apiGroups:
+  - gcp.managed.openshift.io
+  resources:
+  - projectclaims
+  verbs:
+  - get
+  - list
+  - watch
```
Comment (georgettica) [inline deploy/osde2e-tweaks/22-osde2e-noalerts.clusterrole.yaml:21]: Why is this required? Should work without

File: deploy/osde2e-tweaks/non-production/10-noalerts-osde2e.cronjob.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            args:
+            - /bin/bash
+            - -c
+            - |
+              # All CD's that are not "installed"
+              # https://github.com/openshift/ops-sop/blob/master/v4/howto/noalerts.md#hive-stage
+              
+              NAMESPACES=()
+
+              NAMESPACES+=($(oc get cd --all-namespaces -o jsonpath='{.items[?(@.spec.installed!=true)].metadata.namespace}'))
```
Comment (georgettica) [inline deploy/osde2e-tweaks/non-production/10-noalerts-osde2e.cronjob.yaml:32]: Inconsistent naming (following places use the full name)

**Review by georgettica — COMMENTED** [review]
Aside of these minute nuts, fantastic job!
Hope it lands soon
